byway/src/main/webapp/track-interests.js

```diff
@@ -92,6 +93,22 @@ function createButtonForPlace(place) {
   return button;
 }
 
+/* Hide or display list of interests to match user input as a zero-indexed substring. */
+function filterInterests() {
+  const searchField = document.getElementById("search-interest");
+  const searchValue = searchField.value.toLowerCase();
+  const interestBtnsElement = document.getElementById("interests-buttons");
+  let interestBtns = interestBtnsElement.getElementsByClassName("interestBtn");
+  for (let i = 0; i < interestBtns.length; i++) {
+    const btnValue = interestBtns[i].innerText;
+    if (btnValue.toLowerCase().indexOf(searchValue) === 0) {
```

Comment:
If you check `> 0` rather than `=== 0`, you'd be able to search for a partial name, rather than only a prefix.

Reply:
I originally had `> -1`, so that it could also account for being equal to 0, but changed it to the prefix only since typing in `a` still showed a result like `city hall` despite it not starting with`a`. You have a point about it being a partial name, i.e. searching `park` should reveal `RV park`, so I updated despite the match being anywhere in the word.

I could update it to split complete words, i.e. an interests delimited by a space such as `RV park`, and then search for a prefix per beginning of each word. But, this introduces a nested for loop when only using `> -1` accomplishes the same task, albeit not as strictly. I will leave it as is with `> -1` and if you/Landon recommend the split/for loop approach, or some other approach, I can implement.

Reply:
This is an entirely subjective question about how you want your search/filter feature to work. I personally favor the partial match over the prefix approach. This seems more flexible to me. I don't think it is an issue if a search for a single letter has many results. I think it is assumed that the user will type more details in order to get more specific results. E.g. They may type "T" get too many results, and then continue typing "Thea" for Theater  etc. 

You could also split by word, which is better than searching for a strict substring. This is slightly more flexible and would be a better user experience. I wouldn't worry about the nested loop aspect. It is pretty trivial to split/ parse such simple strings. Though in this case I think it is better to stick with the more general approach of checking any partial matches over just prefixes